src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValueNewJson.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Values
+{
+    // Marshalling from Json. These are convenience operations. 
+    public partial class FormulaValue
+    {
+        /// <summary>
+        /// Convenience method to create a value from a json representation. 
+        /// </summary>
+        /// <param name="jsonString"></param>
+        public static FormulaValue FromJson(string jsonString)
+        {
+            try
+            {
+                using var document = JsonDocument.Parse(jsonString);
+                using var jsonMemStream = new MemoryStream();
+                using var paJsonWriter = new Utf8JsonWriter(jsonMemStream);
+                var propBag = document.RootElement;
+
+                return FromJson(propBag);
+            }
+            catch
+            {
+                // $$$ better error handling here?
+                throw;
+            }
+        }
+
+        public static FormulaValue FromJson(JsonElement element)
+        {
+            switch (element.ValueKind)
+            {
+                case JsonValueKind.Null:
+                    return new BlankValue(IRContext.NotInSource(FormulaType.Blank));
+
+                case JsonValueKind.Number:
+                    return new NumberValue(IRContext.NotInSource(FormulaType.Number), element.GetDouble());
+
+                case JsonValueKind.String:
+                    return new StringValue(IRContext.NotInSource(FormulaType.String), element.GetString());
+
+                case JsonValueKind.False:
+                    return new BooleanValue(IRContext.NotInSource(FormulaType.Boolean), false);
+
+                case JsonValueKind.True:
+                    return new BooleanValue(IRContext.NotInSource(FormulaType.Boolean), true);
+
+                case JsonValueKind.Object:
+                    return RecordFromJsonObject(element);
+
+                case JsonValueKind.Array:
+                    return TableFromJsonArray(element);
+
+                default:
+                    throw new NotImplementedException($"Unrecognized JsonElement {element.ValueKind}");
+            }
+        }
+
+        // Json objects parse to records. 
+        internal static RecordValue RecordFromJsonObject(JsonElement element)
```

Comment:
>internal [](http://example.com/codeflow?start=8&length=8)

`private`? This is already exposed via `FromJson(JsonElement)`. Same for TableFromJsonArray